File: BarberApp.Backend/BarberApp.DOMAIN/Models/Checks.cs
using MongoDB.Bson.Serialization.Attributes;


namespace BarberApp.Domain.Models
{
    public class Checks
    {
        [BsonElement("title")]
        public string Title { get; set; }
        [BsonElement("listChecks")]
        public List<Check> ListChecks { get; set; }
        [BsonElement("observation")]
        public string Observation { get; set; }
    }
    public class Check
    {
        [BsonElement("name")]
        public string Name { get; set; }
        [BsonElement("isChecked")]
        public bool IsChecked { get; set; }
    }
}

File: BarberApp.Frontend/src/app/Models/Checks.ts
import { GlobalVariables } from "../Helpers/GlobalVariables"

export interface Checks {
  title: string,
  listChecks: [
    {
      name: string,
      isChecked: boolean
    }
  ],
  observation: string
}

export class ChecksUtilities {
  getDefaultChecks() {
    const companyChecks: Checks = GlobalVariables.companyUserConfig?.checks ?? {
      title: 'Título do grupo',
      listChecks: [
        {
          name: 'Nome do item',
          isChecked: false,
        }
      ],
      observation: ''
    };

    return companyChecks;
  }
}
